core/modules/inputmanager.js

```diff
@@ -0,0 +1,72 @@
+/*\
+title: $:/core/modules/inputmanager.js
+type: application/javascript
+module-type: global
+
+Input handling utilities
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+function InputManager(options) {
+	options = options || "";
+	this.inputs = [];
+}
+
+InputManager.prototype.findInputById = function(identifier) {
+	var result;
+	for(var i=0; i<this.inputs.length; i++) {
+		var object = this.inputs[i];
+		if(object["id"] === identifier) {
+			result = object;
+			break;
+		}
+	}
+	return result;
+};
+
+InputManager.prototype.shouldFocusAgain = function(identifier) {
+	var object = this.findInputById(identifier);
+	if(object && object.shouldFocusAgain && $tw.wiki.getTiddlerText("$:/state/current-focus") === identifier) {
```

Comment:
hmm. `$tw.wiki` is the global wiki instance. ... imo it should be like this: `this.wiki.getTiddlerText(STATE_CURRENT_FOCUS)`